### PR TITLE
Remove destination requirement

### PR DESCRIPTION
**Pull request by pietbrauer**
This removes the `destination` argument from the build as it was before #87. The build was failing and we added it to make it succeed. This made it troublesome in coherence with Carthage.

The solution is to set the `PRODUCT_NAME` and `MODULEMAP_FILE` in the project.

### CODE REVIEW

**Comment by pietbrauer** [comment]
This is so strange. It seems to succeed when it wants to succeed and fail when it is in mood for it. The [Facebook build](https://travis-ci.org/facebook/ios-snapshot-test-case/builds/66166987) succeeded while [my own build](https://travis-ci.org/nerdishbynature/ios-snapshot-test-case/builds/66166985) (same commit) is failing.

**Comment by nscoding** [comment]
@pietbrauer did you figure out why is failing locally?

File: FBSnapshotTestCase/FBSnapshotTestCase.modulemap [hunk 1]
```diff
@@ -0,0 +1,6 @@
+framework module FBSnapshotTestCase {
+    umbrella header "FBSnapshotTestCase.h"
+
+    export *
+    module * { export * }
```
Comment (nscoding) [inline FBSnapshotTestCase/FBSnapshotTestCase.modulemap]: Can you elaborate on this? or point me to docs?
How is that affecting the destination build? :mag:

**Comment by pietbrauer** [comment]
@nscoding Is it failing for you locally? I was only able to reproduce it on Travis. For me locally it worked every single time. Also I am running this branch with CircleCI and Carthage and it builds fine every time. But I agree on not merging it if it is flaky.

File: FBSnapshotTestCase/FBSnapshotTestCase.modulemap (hunk 1 above)
Comment (pietbrauer) [inline FBSnapshotTestCase/FBSnapshotTestCase.modulemap]: TBH I copied this module map from the generated one from CocoaPods. Also @segiddens [pointed out](https://twitter.com/segiddins/status/606574460603375616) I should add a module map.

File: FBSnapshotTestCase.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -345,7 +349,7 @@
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
 				CLANG_WARN_UNREACHABLE_CODE = YES;
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
-				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Distribution";
```
Comment (nscoding) [inline FBSnapshotTestCase.xcodeproj/project.pbxproj]: Was this change required?

Comment (pietbrauer) [inline FBSnapshotTestCase.xcodeproj/project.pbxproj]: Yes, it's Xcode default to use the Distribution profile for Release config.

File: FBSnapshotTestCase/FBSnapshotTestCase.modulemap (hunk 1 above)
Comment (mrackwitz) [inline FBSnapshotTestCase/FBSnapshotTestCase.modulemap]: @pietbrauer: You'd only need a modulemap, if you want to access from Swift a specific part of the Objective-C interface, which shouldn't be part of the regular 'public' API. All involved Objective-C headers must be public though, but you can avoid at least that they are included by default.

File: FBSnapshotTestCase.xcodeproj/project.pbxproj (hunk 2 above)
Comment (nscoding) [inline FBSnapshotTestCase.xcodeproj/project.pbxproj]: :+1:

**Comment by pietbrauer** [comment]
@nscoding Build is stable now with Xcode 6.4 and this is ready for Review/Merge. :fireworks:

**Comment by nscoding** [comment]
@pietbrauer so good! Do you want to squash the commit into one?

**Comment by pietbrauer** [comment]
I rebased into logical chunks but I am fine with squashing.

**Comment by nscoding** [comment]
@pietbrauer your are right, they look very well split. I will merge when I get to the office.

**Comment by pietbrauer** [comment]
@nscoding Super cool!

**Comment by nscoding** [comment]
@pietbrauer this broke [CI](https://travis-ci.org/facebook/ios-snapshot-test-case/builds/69365104) any ideas?

**Comment by pietbrauer** [comment]
Dammit. It ran through like 4-5 times already and now it broke. I would recommend reverting the merge.
